cluster/plan.go

```diff
@@ -466,7 +467,7 @@ func (c *Cluster) BuildKubeletProcess(host *hosts.Host, serviceOptions v3.Kubern
 	Command := c.getRKEToolsEntryPoint(host.OS(), "kubelet")
 	CommandArgs := map[string]string{
 		"client-ca-file": pki.GetCertPath(pki.CACertName),
-		"cloud-provider": c.CloudProvider.Name,
+		"cloud-provider": getCloudProviderName(c.CloudProvider.Name),
```

Comment:
[K8s docs](https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/#:~:text=%2D%2Dcloud%2Dprovider%20string) say the field will be removed from kubelet in k8s 1.24 or later.   
How will the removal affect us, especially when providing a cluster with k8s 1.24 and later?

Reply:
@jiaqiluo It still isn't removed and continues to work with all cloud providers. For external providers, Kubernetes still expects this flag to be set. 

> `kubelet`, `kube-apiserver` and `kube-controller-manager` must be set according to the user's usage of external CCM. If the user has an external CCM (not the internal cloud controller loops in the Kubernetes Controller Manager), then `--cloud-provider=external` must be specified. Otherwise, it should not be specified.

https://kubernetes.io/docs/tasks/administer-cluster/running-cloud-controller/